### PR TITLE
Move yaml config to /hd/config.yaml

### PR DESCRIPTION
**Pull request by alokmenghrajani**
Having the config on /hd/ simplifies the code and makes debugging
easier.

### CODE REVIEW

File: java/gui/src/main/java/com/squareup/subzero/SubzeroConfig.java [hunk 1]
```diff
@@ -23,67 +23,24 @@
...
-      List<String> fileContent =
-          Files.readAllLines(Paths.get(parameter.substring(FILE_PREFIX.length())));
-      if (fileContent.size() > 0) {
-        return fileContent.get(0);
-      }
-    }
-    return parameter;
-  }
-
-  public String getSoftcardPassword() throws IOException {
```
Comment (alokmenghrajani) [inline java/gui/src/main/java/com/squareup/subzero/SubzeroConfig.java]: we only had some getters. Getting rid of them makes things consistent.

File: java/gui/src/main/java/com/squareup/subzero/SubzeroConfig.java [hunk 2]
```diff
@@ -23,67 +23,24 @@
...
-
-  public static class EnvironmentsMap {
-    public Map<String, Environments> environments;
-
-    enum Environments {
-      development
+    InputStream file;
+    if (Strings.isNullOrEmpty(configFile)) {
+      ClassLoader classLoader = SubzeroGui.class.getClassLoader();
+      URL resource = classLoader.getResource("sample.yaml");
```
Comment (mcpherrinm) [inline java/gui/src/main/java/com/squareup/subzero/SubzeroConfig.java]: should we just make this mandatory to provide a --config?

**Comment by mcpherrinm** [comment]
probably worth noting this gets rid of some of the indirection support in the config

File: java/gui/src/main/java/com/squareup/subzero/SubzeroConfig.java (hunk 2 above)
Comment (alokmenghrajani) [reply java/gui/src/main/java/com/squareup/subzero/SubzeroConfig.java]: I like having the dev case just work.

**Pull request merged**